public/app/features/dashboard-scene/scene/Scopes/ScopesFiltersScene.tsx

```diff
@@ -0,0 +1,211 @@
+import { isEqual } from 'lodash';
+import React from 'react';
+import { from, Subscription } from 'rxjs';
+
+import { Scope } from '@grafana/data';
+import {
+  SceneComponentProps,
+  SceneObjectBase,
+  SceneObjectState,
+  SceneObjectUrlSyncConfig,
+  SceneObjectUrlValues,
+  SceneObjectWithUrlSync,
+} from '@grafana/scenes';
+
+import { ScopesFiltersAdvancedSelector } from './ScopesFiltersAdvancedSelector';
+import { ScopesFiltersBasicSelector } from './ScopesFiltersBasicSelector';
+import { fetchNodes, fetchScope, fetchScopes } from './api/scopes';
+import { NodesMap } from './types';
+
+export interface ScopesFiltersSceneState extends SceneObjectState {
+  nodes: NodesMap;
+  loadingNodeName: string | undefined;
+  scopes: Scope[];
+  dirtyScopeNames: string[];
+  isLoadingScopes: boolean;
+  isBasicOpened: boolean;
+  isAdvancedOpened: boolean;
+}
+
+export class ScopesFiltersScene extends SceneObjectBase<ScopesFiltersSceneState> implements SceneObjectWithUrlSync {
+  static Component = ScopesFiltersSceneRenderer;
+
+  protected _urlSync = new SceneObjectUrlSyncConfig(this, { keys: ['scopes'] });
+
+  private fetchNodesSub: Subscription | undefined;
+  private fetchScopesSub: Subscription | undefined;
+
+  constructor() {
+    super({
+      nodes: {
+        '': {
+          name: '',
+          nodeType: 'container',
+          title: '',
+          isExpandable: true,
+          isSelectable: false,
+          isExpanded: true,
+          query: '',
+          nodes: {},
+        },
+      },
+      loadingNodeName: undefined,
+      scopes: [],
+      dirtyScopeNames: [],
+      isLoadingScopes: false,
+      isBasicOpened: false,
+      isAdvancedOpened: false,
+    });
+
+    this.addActivationHandler(() => {
+      this.fetchBaseNodes();
+
+      return () => {
+        this.fetchNodesSub?.unsubscribe();
+        this.fetchScopesSub?.unsubscribe();
+      };
+    });
+  }
+
+  public getUrlState() {
+    return { scopes: this.getScopeNames() };
+  }
+
+  public updateFromUrl(values: SceneObjectUrlValues) {
+    let dirtyScopeNames = values.scopes ?? [];
+    dirtyScopeNames = Array.isArray(dirtyScopeNames) ? dirtyScopeNames : [dirtyScopeNames];
+
+    this.updateScopes(dirtyScopeNames);
+  }
+
+  public fetchBaseNodes() {
+    return this.updateNode([''], true, '');
+  }
+
+  public updateNode(path: string[], isExpanded: boolean, query: string) {
+    let nodes = { ...this.state.nodes };
+    let currentLevel: NodesMap = nodes;
+
+    for (let idx = 0; idx < path.length - 1; idx++) {
+      currentLevel = currentLevel[path[idx]].nodes;
+    }
+
+    const name = path[path.length - 1];
+    const currentNode = currentLevel[name];
+
+    if (isExpanded || currentNode.query !== query) {
+      this.fetchNodesSub?.unsubscribe();
+
+      this.setState({ loadingNodeName: name });
+
+      this.fetchNodesSub = from(fetchNodes(name, query)).subscribe((childNodes) => {
```

Comment:
Similar to my other comment, do we really need a subscription to handle simple async action?